vespalib/src/vespa/vespalib/net/crypto_engine.cpp

```diff
@@ -155,10 +155,29 @@ class XorCryptoSocket : public CryptoSocket
     }
 };
 
+CryptoEngine::SP create_default_crypto_engine() {
+    // TODO: check VESPA_TLS_CONFIG_FILE here
+    // return std::make_shared<XorCryptoEngine>();
+    return std::make_shared<NullCryptoEngine>();
 }
 
+} // namespace vespalib::<unnamed>
+
+std::mutex CryptoEngine::_shared_lock;
+CryptoEngine::SP CryptoEngine::_shared_default(nullptr);
+
 CryptoEngine::~CryptoEngine() = default;
 
+CryptoEngine::SP
+CryptoEngine::get_default()
+{
+    std::lock_guard<std::mutex> guard(_shared_lock);
```

Comment:
Can simplify to `std::lock_guard guard(_shared_lock)` due to C++17 auto template parameter type inference 🦄✨

Reply:
oooh nice; will start to use this feature